packages/plugin-ext/src/main/browser/menus/menus-contribution-handler.ts

```diff
@@ -59,36 +67,73 @@ export class MenusContributionPointHandler {
                         this.quickCommandService.pushCommandContext(menu.command, menu.when);
                     }
                 }
+            } else if (location === 'editor/title') {
+                this.registerEditorTitleActions(allMenus[location], contributions);
             } else if (allMenus.hasOwnProperty(location)) {
-                const menuPath = MenusContributionPointHandler.parseMenuPath(location);
-                if (!menuPath) {
+                const menuPaths = MenusContributionPointHandler.parseMenuPaths(location);
+                if (!menuPaths.length) {
                     this.logger.warn(`Plugin contributes items to a menu with invalid identifier: ${location}`);
                     continue;
                 }
                 const menus = allMenus[location];
                 menus.forEach(menu => {
-                    if (!this.isMenuItemRegistered(location, menu.command)) {
-                        this.registerMenuAction(menuPath, location, menu);
+                    for (const menuPath of menuPaths) {
+                        this.registerMenuAction(menuPath, menu);
                     }
                 });
             }
         }
     }
 
-    protected static parseMenuPath(value: string): MenuPath | undefined {
-        switch (value) {
-            case 'editor/context': return EDITOR_CONTEXT_MENU;
-            case 'explorer/context': return NAVIGATOR_CONTEXT_MENU;
-            case 'view/item/context': return VIEW_ITEM_CONTEXT_MENU;
+    protected registerEditorTitleActions(actions: Menu[], contributions: PluginContribution): void {
+        if (!contributions.commands || !actions.length) {
+            return;
+        }
+        const commands = new Map(contributions.commands.map(c => [c.command, c] as [string, PluginCommand]));
+        for (const action of actions) {
+            const pluginCommand = commands.get(action.command);
+            if (pluginCommand) {
+                this.registerEditorTitleAction(action, pluginCommand);
+            }
         }
     }
 
-    protected isMenuItemRegistered(location: string, commandId: string): boolean {
-        const commands = this.registeredMenus.get(location);
-        return commands !== undefined && commands.has(commandId);
+    protected editorTitleActionId = 0;
+    protected registerEditorTitleAction(action: Menu, pluginCommand: PluginCommand): void {
+        const id = pluginCommand.command;
+        const command = '__editor.title.' + id;
+        const tooltip = pluginCommand.title;
+        const iconClass = 'plugin-editor-title-action-' + this.editorTitleActionId++;
+        const { group, when } = action;
+
+        const { iconUrl } = pluginCommand;
+        const darkIconUrl = typeof iconUrl === 'object' ? iconUrl.dark : iconUrl;
```

Comment:
how it will handle multiple themes ? if we assume it's either dark or white ?
https://code.visualstudio.com/docs/getstarted/themes

Reply:
I've not tried to implement it theming, but only aligned with how VS Code register icons for commands to support them in the editor title menu.

Looking at the code it does not seem that VS Code supports theming of command icons:
https://github.com/Microsoft/vscode/blob/e83cefd1e79f3082825c72c633434e99d09f9104/src/vs/workbench/api/common/menusExtensionPoint.ts#L173-L180

Reply:
I still don't see then why there is a need of "light" and "dark" there (but moved the question upper in the code)